ibis/expr/datatypes.py

```diff
@@ -243,85 +263,103 @@ class Floating(Primitive):
     __slots__ = ()
 
     @property
-    def largest(self):
+    def largest(self) -> 'Float64':
         return float64
 
+    @property
+    def nbytes(self) -> int:
+        raise TypeError(
+            'Cannot compute the number of bytes of an abstract floating point '
+            'type'
+        )
 
-class Int8(SignedInteger):
 
+class Int8(SignedInteger):
     __slots__ = ()
 
-    _nbytes = 1
+    @property
+    def nbytes(self) -> int:
+        return 1
 
 
 class Int16(SignedInteger):
-
     __slots__ = ()
 
-    _nbytes = 2
+    @property
+    def nbytes(self) -> int:
+        return 2
 
 
 class Int32(SignedInteger):
-
     __slots__ = ()
 
-    _nbytes = 4
+    @property
+    def nbytes(self) -> int:
+        return 4
 
 
 class Int64(SignedInteger):
-
     __slots__ = ()
 
-    _nbytes = 8
+    @property
+    def nbytes(self) -> int:
+        return 8
 
 
 class UInt8(UnsignedInteger):
-
     __slots__ = ()
 
-    _nbytes = 1
+    @property
+    def nbytes(self) -> int:
+        return 1
 
 
 class UInt16(UnsignedInteger):
-
     __slots__ = ()
 
-    _nbytes = 2
+    @property
+    def nbytes(self) -> int:
+        return 2
 
 
 class UInt32(UnsignedInteger):
-
     __slots__ = ()
 
-    _nbytes = 4
+    @property
+    def nbytes(self) -> int:
+        return 4
 
 
 class UInt64(UnsignedInteger):
-
     __slots__ = ()
 
-    _nbytes = 8
+    @property
+    def nbytes(self) -> int:
+        return 8
 
 
 class Halffloat(Floating):
```

Comment:
I'd rename this to `Float16` and have a `Halffloat` alias not the other way around.